test/date.test.js

```diff
@@ -58,6 +58,24 @@ test('render a date in a string when format is date as YYYY-MM-DD', (t) => {
   t.ok(validate(JSON.parse(output)), 'valid schema')
 })
 
+test('render date in GMT+1 time zone correctly as YYYY-MM-DD', (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'a date in a string',
+    type: 'string',
+    format: 'date'
+  }
+  const toStringify = new Date('August 19, 1975 00:15:30 GMT+01:00')
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify(toStringify)
+
+  t.equal(output, '"1975-08-19"')
```

Comment:
CI is failing

```
2021-09-14T16:00:05.9090566Z             --- expected
2021-09-14T16:00:05.9090925Z             +++ actual
2021-09-14T16:00:05.9091357Z             @@ -1,1 +1,1 @@
2021-09-14T16:00:05.9091956Z             -"1975-08-19"
2021-09-14T16:00:05.9092379Z             +"1975-08-18"
```

My TZ is GMT+2, it should be related

Reply:
The result from the formatting is equal to what `moment(toStringify).format('YYYY-MM-DD')` would output.

Instantiating a new Date automatically converts it to UTC.

`getTimezoneOffset` gets the offset compared to the machine running the code, thus yielding different results.

The current implementation takes that into account to always output the UTC date and not put out any system-specific dates (otherwise the other tests fail locally) - this is the same behavior as moment.js.

Tbh, I'd keep this test out and keep the implementation as is.

WDYT?

Reply:
> CI is failing
> 
> ```
> 2021-09-14T16:00:05.9090566Z             --- expected
> 2021-09-14T16:00:05.9090925Z             +++ actual
> 2021-09-14T16:00:05.9091357Z             @@ -1,1 +1,1 @@
> 2021-09-14T16:00:05.9091956Z             -"1975-08-19"
> 2021-09-14T16:00:05.9092379Z             +"1975-08-18"
> ```
> 
> My TZ is GMT+2, it should be related

That's what was wondering. We need to enforce UTC ain't we?

Reply:
@zekth We are enforcing UTC in the implementation. From your comment

> console.log(s(new Date('August 19, 1975 00:15:30 GMT+01:00'))) // "1975-08-19"

I thought, you want to the result of the formatting from that date to be 1975-08-19, which wouldn't be UTC and would be different to the current `Intl` implementation.

Do you want to remove that new test or do you want the assertion to be 1975-08-18?

Reply:
Assertion shouldn't be changed. Atm i have no proper solution that comes to my mind.

Reply:
Again, this would break the current implementation. It does not work with JS dates like that. That assertion would also break with the current `Intl` implementation. The test would fail without my changes given that assertion. I don't think we are trying to introduce a breaking change here.